src/components/Fieldset/Fieldset.jsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames/bind';
+import styles from './Fieldset.scss';
+const cx = classnames.bind( styles );
+
+const Fieldset = ( {
+	autoComplete,
+	defaultValue,
+	describedBy,
+	disabled,
+	isMultiline,
+	inputRef,
+	id,
+	label,
+	maxLength,
+	placeholder,
+	onBlur,
+	onChange,
+	onClick,
+	onFocus,
+	onInvalid,
+	pattern,
+	readOnly,
+	required,
+	type,
+} ) => {
+	// ClassNames used by both <input> and <textarea>.
+	const sharedClassNames = {
+		disabled,
+		readOnly,
+		[type]: type,
+	};
+
+	const requiredField = label && required;
+
+	return (
+		<fieldset className={styles.container}>
```

Comment:
I don't know that fieldset is quite right here. My understanding is that you would use this to group semantically related inputs, e.g. radio buttons like Yes, No, Maybe. It's by no means an invalid use of the element, but it does seem a bit opinionated and something that a consumer may want the option to use themselves. To my mind, label is a perfectly good outer element for this component.